File: spotify-browser-frontend/src/app/core/paging.ts
export interface IPaging<T> {
    href: string;
    items: T[];
    limit: number;
    offset: number;
    next: string;
    previous?: any;
    total: number;
}


export class Paging<T> implements IPaging<T> {
    href: string;
    items: T[];
    limit: number;
    offset: number;
    next: string;
    previous?: any;

    private _total: number;

    get total(): number {
        if (this._total > 1000) {
          return 1000;
        }
        return this._total;
      }

    constructor( pagedList: IPaging<any>, itemMapper: (any: any) => T ) {
        this.href = pagedList.href;
        this.limit = pagedList.limit;
        this.offset = pagedList.offset;
        this.next = pagedList.next;
        this.previous = pagedList.previous;
        this._total = pagedList.total;
        this.items = pagedList.items.map( i => itemMapper(i));
    }
}

File: SpotifyBrowser.Backend/SpotifyBrowser.Cqrs.Implementation/Paging.cs
using System.Collections.Generic;
using Microsoft.AspNetCore.Routing;

namespace SpotifyBrowser.Cqrs.Implementation
{
    public class Paging<T>
    {
        public string Href { get; set; }
        public IEnumerable<T> Items { get; set; }
        public int Limit { get; set; }
        public int Offset { get; set; }
        public string Next { get; set; }
        public string Previous { get; set; }
        public int Total { get; set; }

        public Paging(IEnumerable<T> items, int limit, int offset, int total, string href = "", string next = "" , string previous = "" )
        {
            Items = items;
            Limit = limit;
            Offset = offset;
            Total = total;
            Href = href;
            Next = next;
            Previous = previous;
        }
    }
}
